bgpd/bgp_evpn.c

```diff
@@ -1685,6 +1685,55 @@ static int update_evpn_route_entry(struct bgp *bgp, struct bgpevpn *vpn,
 	return route_change;
 }
 
+/*
+ * if the local route was not selected evict it and tell zebra
+ * to add the best remote dest
+ */
+static void evpn_cleanup_local_non_best_route(struct bgp *bgp,
+					      struct bgpevpn *vpn,
+					      struct bgp_node *rn,
+					      struct bgp_path_info *local_pi,
+					      int *route_change)
+{
+	struct bgp_path_info *tmp_pi;
+	struct bgp_path_info *curr_select = NULL;
+	uint8_t flags = 0;
+	char buf[PREFIX_STRLEN];
+
+	if (CHECK_FLAG(local_pi->flags, BGP_PATH_SELECTED)) {
```

Comment:
Should we have this check before calling this function ?

Reply:
sure, that is probably more optimal. will fix it up.